graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java

```diff
@@ -92,6 +93,7 @@ public EventDefinitionsResource(DBEventDefinitionService dbService,
     public PaginatedResponse<EventDefinitionDto> list(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
                                                       @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("50") int perPage,
                                                       @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query) {
+        checkPermission(RestPermissions.EVENT_DEFINITIONS_READ);
```

Comment:
We have to check for each event definition ID here since our permission system can check up to the single entity.
In the database service we can use `findPaginatedWithQueryFilterAndSort()` with a `Predicate` to filter the event definitions with permissions.

Example:

https://github.com/Graylog2/graylog2-server/blob/9f38dd10366f970134c2b6225ec3edbc23f1319b/graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/ViewsResource.java#L116-L121